### DISCUSSION TITLE
Filter API on M2M with "_neq" won't return records that has empty array

### DISCUSSION THREAD

**Discussion by Nierrrrrrr** [directus/directus#9922] [Archive]
Hi,

I have A record with a M2M to B record and I want to find that all A records which doesn't related to specified B record.

For example, I want to find all Students (A record) that doesn't take one specified Class (B record) like "Math", I tried the filter API like this:

```
student(filter: {
  class: {
    class_list_id: {
      class_name: {
        _neq: "Math"
      }
    }
  }
})
```

If students take other classes except "Math", they will be returned. However, students that doesn't take any classes won't be returned. How can I set the filter that will includes those students without taking any class?

**Comment by rijkvanzanten** [comment]
There's currently no filter that checks against the amount of related items in a relational field
